pubsub/pubsub.go

```diff
@@ -424,9 +425,12 @@ func (s *Subscription) Receive(ctx context.Context) (_ *Message, err error) {
 		// Even though the mutex is unlocked, only one goroutine can be here.
 		// The only way here is if s.waitc was nil. This goroutine just set
 		// s.waitc to non-nil while holding the lock.
+		if s.preReceiveBatchHook != nil {
+			s.preReceiveBatchHook(nMessages)
+		}
 		msgs, err := s.getNextBatch(ctx, nMessages)
-		if s.onReceiveBatchHook != nil {
-			s.onReceiveBatchHook(len(msgs), nMessages)
+		if s.postReceiveBatchHook != nil {
+			s.postReceiveBatchHook(len(msgs))
```

Comment:
Maybe this should take the error too, so it can decide what it wants to do if err != nil.

Reply:
I'll add it if needed.